database/database.go

```diff
@@ -170,6 +185,103 @@ func (db SqlDatabase) AddImage(uuid string, name string, alt string) (err error)
 	return nil
 }
 
+func (db SqlDatabase) AddCard(uuid string, image_location string, json_data string, schema_name string) (err error) {
+	// Check that the file exists and is a file
+	// not a directory. Ideally, check the ext
+	if image_location == "" {
+		return fmt.Errorf("cannot have image")
+	}
+	image_stat, err := os.Stat(image_location)
+	if errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("file does not exist: %s", image_location)
+	}
+	if err != nil {
+		return err
+	}
+	if image_stat.IsDir() {
+		return fmt.Errorf("given path is a directory: %s", image_location)
+	}
+
+	// Load the schema
+	// TODO : probably pass the schema data instead
+	if json_data == "" {
+		return fmt.Errorf("cannot have empty data")
+	}
+
+	if schema_name == "" {
+		return fmt.Errorf("cannot have an empty schema name")
+	}
+
+	_, err = validateJson(json_data, schema_name)
+	if err != nil {
+		return err
+	}
+
+	// Insert everything to the DB
+	tx, err := db.Connection.Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if commit_err := tx.Commit(); commit_err != nil {
+			err = errors.Join(err, tx.Rollback(), commit_err)
+		}
+	}()
+
+	query := "INSERT INTO cards(uuid, image_location, json_data, json_schema) VALUES(?, ?, ?, ?);"
+	_, err = tx.Exec(query, uuid, image_location, json_data, schema_name)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// / This function gets a post from the database
+// / with the given ID.
+func (db SqlDatabase) GetCard(uuid int) (card common.Card, err error) {
+	rows, err := db.Connection.Query("SELECT image_location, json_data, json_schema FROM cards WHERE uuid=?;", uuid)
+	if err != nil {
+		return common.Card{}, err
+	}
+	defer func() {
+		err = errors.Join(rows.Close())
+	}()
+
+	rows.Next()
+	if err = rows.Scan(&card.ImageLocation, &card.JsonData, &card.SchemaName); err != nil {
+		return common.Card{}, err
+	}
+
+	// Validate the json
+	validateJson(card.JsonData, card.SchemaName)
```

Comment:
Is there a reason why you need to validate the data after loading it from the database? If the data was validated when storing it to the database then validating it again should be obsolete, unless the configured schema has changed. And by this point the data in the database is broken and should be deleted anyways.